components/atom/imagePanel/src/index.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import classnames from 'classnames'
+
+const BASE_16 = 16
+const HORIZONTAL_ALIGNMENTS = {
+  LEFT: 'left',
+  CENTER: 'center',
+  RIGHT: 'right'
+}
+
+const VERTICAL_ALIGNMENTS = {
+  TOP: 'top',
+  CENTER: 'center',
+  BOTTOM: 'bottom'
+}
+
+const TYPES = {
+  RESIZED: 'resized'
+}
+
+const hexToRgb = function (hex) {
+  const shorthandRegex = /^#?([a-f\d])([a-f\d])([a-f\d])$/i
+  hex = hex.replace(shorthandRegex, (m, r, g, b) => r + r + g + g + b + b)
+
+  var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex)
+  return result ? {
+    r: parseInt(result[1], BASE_16),
+    g: parseInt(result[2], BASE_16),
+    b: parseInt(result[3], BASE_16)
+  } : {}
+}
+
+const getClassNames = function ({verticalAlign, horizontalAlign, classNames, resized}) {
+  return classnames(
+    'sui-AtomImagePanel',
+    `sui-AtomImagePanel--v-${verticalAlign}`,
+    `sui-AtomImagePanel--h-${horizontalAlign}`,
+    resized && `sui-AtomImagePanel-resized`,
+    classNames
+  )
+}
+
+const getStyles = function ({placeholderColor, overlayColor, src, overlayAlpha}) {
+  const url = `url(${src})`
+  if (overlayColor) {
+    const overlayColorRgb = hexToRgb(overlayColor)
+    const rgba = `rgba(${overlayColorRgb.r}, ${overlayColorRgb.g}, ${overlayColorRgb.b}, ${overlayAlpha})`
```

Comment:
Do we really to add a gradient that is not a gradient but background with alpha color ? 
I get that we're trying to minimize the number of DOM nodes. But, are we measuring which is better between one more nodes or repainting a gradient twice ?
Have you consider using Before or After pseudo elements ?

Reply:
Finally we will use the :: before pseudo element